test/e2e/apiextensions_test.go

```diff
@@ -377,6 +378,51 @@ func TestPropagateFieldsRemovalToXRAfterUpgrade(t *testing.T) {
 	)
 }
 
+// TestPropagateFieldsRemovalToComposed tests Crossplane's end-to-end SSA syncing
+// functionality of clear propagation of fields from claim->XR->MR, when existing
+// composition and resources are migrated from native P-and-T to functions pipeline mode.
+func TestPropagateFieldsRemovalToComposed(t *testing.T) {
+	manifests := "test/e2e/manifests/apiextensions/composition/propagate-field-removals"
+	environment.Test(t,
+		features.New(t.Name()).
+			WithLabel(LabelArea, LabelAreaAPIExtensions).
+			WithLabel(LabelSize, LabelSizeSmall).
+			WithLabel(LabelModifyCrossplaneInstallation, LabelModifyCrossplaneInstallationTrue).
+			WithLabel(config.LabelTestSuite, SuiteSSAClaims).
+			WithSetup("EnableSSAClaims", funcs.AllOf(
+				funcs.AsFeaturesFunc(environment.HelmUpgradeCrossplaneToSuite(SuiteSSAClaims)),
+				funcs.ReadyToTestWithin(1*time.Minute, namespace),
+			)).
+			WithSetup("PrerequisitesAreCreated", funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "setup/*.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "setup/*.yaml"),
+				funcs.ResourcesHaveConditionWithin(1*time.Minute, manifests, "setup/definition.yaml", apiextensionsv1.WatchingComposite()),
+			)).
+			Assess("CreateClaim", funcs.AllOf(
+				funcs.ApplyClaim(FieldManager, manifests, "claim.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "claim.yaml"),
+				funcs.ResourcesHaveConditionWithin(5*time.Minute, manifests, "claim.yaml", xpv1.Available()),
+			)).
+			Assess("ConvertToPipelineCompositionUpgrade", funcs.ApplyResources(FieldManager, manifests, "composition-xfn.yaml")).
+			Assess("UpdateClaim", funcs.ApplyClaim(FieldManager, manifests, "claim-update.yaml")).
+			Assess("FieldsRemovalPropagatedToMR",
+				// field removals are propagated claim -> XR -> MR, after converting composition from native to pipeline mode
+				funcs.ComposedResourcesHaveFieldValueWithin(1*time.Minute, manifests, "claim.yaml",
+					"spec.forProvider.fields.tags[newtag]", funcs.NotFound,
```

Comment:
is it also worth a check that `tags[tag]` still exists, to confirm it wasn't accidentally removed? Up to you if you think that's a useful check to add :)

Reply:
Updated to assert that both field removals and updates (tags[tag)) propagates as expected.